### PR TITLE
add org.firo.firo-qt

### PR DESCRIPTION
**Pull request by erciccione**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: org.firo.firo-qt.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>org.firo.firo-qt</id>
+  <name>Firo</name>
+  <metadata_license>MIT</metadata_license>
+  <summary>
+    Fully validating Firo peer-to-peer network node, wallet and graphical interface.
```
Comment (bbhtt) [inline org.firo.firo-qt.metainfo.xml:7]:
```suggestion
    Fully validating Firo peer-to-peer network node, wallet and graphical interface
```

File: org.firo.firo-qt.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>org.firo.firo-qt</id>
+  <name>Firo</name>
+  <metadata_license>MIT</metadata_license>
```
Comment (bbhtt) [inline org.firo.firo-qt.metainfo.xml:5]: Missing project license https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#license

File: org.firo.firo-qt.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+Bitcoin was originally created as an answer to this by ensuring you can be self sovereign over your money and to serve as uncensorable and unseizable money that isn't controlled by any one entity. Bitcoin's lack of privacy however has now made it much easier to seize or blacklist funds and due to ossification of the protocol, is unlikely to take serious steps to address this.
+
+Firo has dedicated itself to being a privacy preserving cryptocurrency and have designed and built trustless privacy protocols such as Lelantus and Lelantus Spark that have inspired and shaped the designs of other privacy protocols (for e.g. Triptych, Seraphis, Lelantus-MW). 
+    </p>
+  </description>
+  <screenshots>
+    <screenshot>
+      <image type="source">
+        https://raw.githubusercontent.com/firoorg/firo/master/contrib/debian/qt-screenshot.jpg
```
Comment (bbhtt) [inline org.firo.firo-qt.metainfo.xml:30]: https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#screenshots

> If the images are hosted in a git repository, the link should be from a tag or a commit and not a branch.

File: org.firo.firo-qt.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      </image>
+    </screenshot>
+  </screenshots>
+  <releases>
+    <release date="2023-12-23" version="0.14.13.1"/>
+  </releases>
+  <content_rating type="oars-1.1"/>
```
Comment (bbhtt) [inline org.firo.firo-qt.metainfo.xml:37]: If this involves transactions it should be correctly tagged with OARS information.

File: org.firo.firo-qt.json [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                    "url": "https://raw.githubusercontent.com/firoorg/firo/master/contrib/debian/firo-qt.desktop",
+                    "sha256": "79070bd3af4334f4239585630d7d7d6b8d417a15ca7c1ba1a4ef6e4bc276bed4"
+                },
+                {
+                    "type": "file",
+                    "path": "org.firo.firo-qt.metainfo.xml"
+                },
+                {
+                    "type": "file",
+                    "url": "https://raw.githubusercontent.com/firoorg/firo/master/src/qt/res/icons/firo.svg",
```
Comment (bbhtt) [inline org.firo.firo-qt.json:63]: From a commit, otherwise sha256 will change randomly

File: org.firo.firo-qt.json [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+                "desktop-file-edit --set-key=\"Icon\" --set-value=${FLATPAK_ID} firo-qt.desktop",
+                "install -Dp -m 644 firo-qt.desktop /app/share/applications/${FLATPAK_ID}.desktop",
+                "install -Dp -m 644 firo.svg /app/share/icons/hicolor/scalable/apps/org.firo.firo-qt.svg",
+                "install -Dp -m 644 org.firo.firo-qt.metainfo.xml /app/share/metainfo/org.firo.firo-qt.metainfo.xml",
+                "bash install_wrappers.sh"
+            ],
+            "sources": [
+                {
+                    "type": "file",
+                    "url": "https://raw.githubusercontent.com/firoorg/firo/master/contrib/debian/firo-qt.desktop",
```
Comment (bbhtt) [inline org.firo.firo-qt.json:54]: From a commit, otherwise sha256 will change randomly

File: org.firo.firo-qt.metainfo.xml
Comment (bbhtt) [inline org.firo.firo-qt.metainfo.xml:1]: This file should be submitted upstream

**Comment by bbhtt** [comment]
> I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. 

How are you associated with the project?

File: org.firo.firo-qt.json [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+                "install -m 0744 -D -t /app/bin/internal bin/firo-qt",
+                "install -m 0744 -D -t /app/bin/internal bin/firo-cli"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "only-arches": [
+                        "x86_64"
+                    ],
+                    "url": "https://github.com/firoorg/firo/releases/download/v0.14.13.1/firo-0.14.13.1-linux64.tar.gz",
```
Comment (hfiguiere) [inline org.firo.firo-qt.json:30]: how does this work on aarch64? An aarch64 flatpak is still built.

why not building from source instead?

File: org.firo.firo-qt.metainfo.xml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    https://firo.org/
+  </url>
+  <url type="bugtracker">
+    https://github.com/firo/firo/issues/new?title=[flatpak]
+  </url>
+  <launchable type="desktop-id">
+    org.firo.firo-qt.desktop
+  </launchable>
+  <description>
+    <p>
```
Comment (hfiguiere) [inline org.firo.firo-qt.metainfo.xml:19]: this will be shown as a single paragraph, like it would be in HTML.

File: org.firo.firo-qt.json [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+                        "type": "html",
+                        "url": "https://github.com/firoorg/firo/releases.atom",
+                        "version-pattern": "<title>Firo ([\\d\\.]+)</title>",
+                        "url-template": "https://github.com/firoorg/firo/releases/download/$version/firo-$version-linux64.tar.gz"
+                    }
+                }
+            ]
+        },
+        {
+            "name": "misc",
```
Comment (hfiguiere) [inline org.firo.firo-qt.json:42]: no reason to make this a separate module.

File: install_wrappers.sh
Comment (hfiguiere) [inline install_wrappers.sh:1]: this has no reason to exist.

There is a `script` source type.

File: org.firo.firo-qt.metainfo.xml (hunk 3 above)
Comment (erciccione) [reply org.firo.firo-qt.metainfo.xml]: Done

File: org.firo.firo-qt.metainfo.xml (hunk 2 above)
Comment (erciccione) [reply org.firo.firo-qt.metainfo.xml:5]: added

File: org.firo.firo-qt.json (hunk 5 above)
Comment (erciccione) [reply org.firo.firo-qt.json]: done

File: org.firo.firo-qt.json (hunk 9 above)
Comment (erciccione) [reply org.firo.firo-qt.json]: moved everything to one single module.

File: install_wrappers.sh
Comment (erciccione) [reply install_wrappers.sh:1]: I tried to recreate the same behaviour using the script source type, but i didn't manage to get the same result. I've seen other repositories using a wrapper script (including the project whose file i've used as a base for this). If it doesn't make any practical difference, i'd prefer to keep it as it is.

File: org.firo.firo-qt.json (hunk 7 above)
Comment (erciccione) [reply org.firo.firo-qt.json:35]: done

File: org.firo.firo-qt.metainfo.xml
Comment (erciccione) [reply org.firo.firo-qt.metainfo.xml:1]: What's upstream in this context? The official repository of the project?

File: org.firo.firo-qt.json (hunk 6 above)
Comment (erciccione) [reply org.firo.firo-qt.json]: done

File: org.firo.firo-qt.metainfo.xml (hunk 4 above)
Comment (erciccione) [reply org.firo.firo-qt.metainfo.xml:39]: The generated tag is still only `<content_rating type="oars-1.1" />`. No additional info needed.

**Comment by erciccione** [comment]
Thank you both for the review.

> How are you associated with the project?

@bbhtt i'm a regular contributor who closely works with the Firo team. I've been authorized to submit the app to flathub and maintain it.

File: install_wrappers.sh
Comment (hfiguiere) [reply install_wrappers.sh:1]: > including the project whose file i've used as a base for this).

two wrongs don't make a right.

File: org.firo.firo-qt.json (hunk 7 above)
Comment (hfiguiere) [reply org.firo.firo-qt.json:35]: what's done? A aarch64 package is still build. I suggested building from source as an alternative.

File: org.firo.firo-qt.json [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+        "--filesystem=/run/tor/control.authcookie:ro"
+    ],
+    "modules": [
+        {
+            "name": "firo",
+            "buildsystem": "simple",
+            "build-commands": [
+                "install -m 0744 -D -t /app/bin/internal bin/firod",
+                "install -m 0744 -D -t /app/bin/internal bin/firo-qt",
+                "install -m 0744 -D -t /app/bin/internal bin/firo-cli",
```
Comment (hfiguiere) [inline org.firo.firo-qt.json:22]:
```suggestion
                "install -m 0755 -D -t /app/bin/internal bin/firod",
                "install -m 0755 -D -t /app/bin/internal bin/firo-qt",
                "install -m 0755 -D -t /app/bin/internal bin/firo-cli",
```

File: install_wrappers.sh
Comment (hfiguiere) [reply install_wrappers.sh:1]: once again, make this inline in the manifest. I don't see what the problem is.

File: org.firo.firo-qt.metainfo.xml
Comment (bbhtt) [reply org.firo.firo-qt.metainfo.xml:1]: Yes

Comment (erciccione) [reply org.firo.firo-qt.metainfo.xml:1]: Done: https://github.com/firoorg/firo/pull/1415

File: install_wrappers.sh
Comment (erciccione) [reply install_wrappers.sh:1]: Done

File: org.firo.firo-qt.json (hunk 10 above)
Comment (erciccione) [reply org.firo.firo-qt.json]: Done

**Comment by erciccione** [comment]
Edited files so we start with the latest version of the app.

File: org.firo.firo-qt.json [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+                "install -Dp -m 644 org.firo.firo-qt.metainfo.xml /app/share/metainfo/org.firo.firo-qt.metainfo.xml",
+                "for b in firo{d,-qt,-cli}; do echo '#!/usr/bin/bash' > /app/bin/$b && echo \"exec /app/bin/internal/$b\" '-datadir=\"${XDG_DATA_HOME}\" \"$@\"' >> /app/bin/$b && chmod 744 /app/bin/$b; done"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "only-arches": [
+                        "x86_64"
+                    ],
+                    "url": "https://github.com/firoorg/firo/releases/download/v0.14.13.2/firo-0.14.13.2-linux64.tar.gz",
```
Comment (hfiguiere) [inline org.firo.firo-qt.json:35]: so what happen in the aarch64 package?

Comment (erciccione) [reply org.firo.firo-qt.json:35]: aarch64 support is not maintained, so i haven't added it. Maybe i'm missing the problem. Should we provide an aarch64 package anyway for flathub even if we don't support it upstream?

Comment (bbhtt) [reply org.firo.firo-qt.json:35]: You are building for both arches right now while shipping a x86_64 only binary https://docs.flathub.org/docs/for-app-authors/maintenance#limiting-the-set-of-architectures-to-build-on

Comment (erciccione) [reply org.firo.firo-qt.json:35]: I missed that a flatpak.json file was also needed containing `only-arches` . Added.

File: org.firo.firo-qt.metainfo.xml [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+    We have seen how freedom of commerce or even access to our savings are things that are no longer things that can be taken for granted and authoritarian governments have increasingly used money as a tool of control.</p>
+
+    <p>As fiat currencies go increasingly digital with some going as far as totally replacing physical cash, we lose control over our own money instead relying on intermediaries. 
+    Bitcoin was originally created as an answer to this by ensuring you can be self sovereign over your money and to serve as uncensorable and unseizable money that isn't controlled by any one entity.</p>
+
+    <p>Bitcoin's lack of privacy however has now made it much easier to seize or blacklist funds and due to ossification of the protocol, is unlikely to take serious steps to address this.
+    Firo has dedicated itself to being a privacy preserving cryptocurrency and have designed and built trustless privacy protocols such as Lelantus and Lelantus Spark that have inspired and shaped the designs of other privacy protocols (for e.g. Triptych, Seraphis, Lelantus-MW).</p>
+  </description>
+  <screenshots>
+    <screenshot>
```
Comment (bbhtt) [inline org.firo.firo-qt.metainfo.xml:30]:
```suggestion
    <screenshot type="default">
```

Comment (erciccione) [reply org.firo.firo-qt.metainfo.xml]: added

**Comment by erciccione** [comment]
@bbhtt anything else needed before this pr can be accepted?

**Comment by bbhtt** [comment]
bot, build org.firo.firo-qt

File: flatpak.json
[the remark this answers is not in the document]
Comment (erciccione) [reply flatpak.json:1]: whoops :) renamed.

**Comment by bbhtt** [comment]
bot, build org.firo.firo-qt

**Pull request closed without merging**